### DISCUSSION TITLE
Enhancing Simplicity and Performance in Headless (glTF) Rendering

### DISCUSSION THREAD

**Discussion by vincent-herlemont** [bevyengine/bevy#13596] [Q&A]
I would like to do a headless glTF rendering. I found a very simple example here: https://bevyengine.org/examples/3D%20Rendering/load-gltf/

However, for headless rendering.

I found this example: [headless_renderer.rs](https://github.com/bevyengine/bevy/blob/ea283c1dead8a1b3d6929d1e9d2d2bdddaa93e05/examples/app/headless_renderer.rs).

The headless renderer example raises some questions for me about its complexity, the need to communicate between the “main world” and the “render world”. Isn’t it possible that at the end of the setup we could take a screenshot without needing multiple “worlds”?  To render the image immediately after loading it, without needing the update system described [here `headless_renderer.rs#L476-L547`](https://github.com/bevyengine/bevy/blob/ea283c1dead8a1b3d6929d1e9d2d2bdddaa93e05/examples/app/headless_renderer.rs#L476-L547).

In other words, is there a simpler way to do a headless renderer?

There might also be a performance concern, as it takes ~2 seconds to generate the image with [this example](https://github.com/bevyengine/bevy/blob/ea283c1dead8a1b3d6929d1e9d2d2bdddaa93e05/examples/app/headless_renderer.rs). Is there a faster way to do it?
